### PR TITLE
Replace SWIG wrappers with pybind11 (DM-11593)

### PR DESCRIPTION
**Pull request by andy-slac**
This replaces qmetaLib and cssLib modules which were generated by SWIG with pybind11 hand-coded stuff. This does not remove dependency on SWIG yet as it is used for LUA bindings too, we'll work on LUA on separate ticket.

### CODE REVIEW

File: core/modules/css/SConscript [hunk 1]
```diff
@@ -8,20 +8,15 @@ Import('standardModule')
...
-# Python module is generated in python/ directory for standardModule()
-swigged = env.SharedObject(source="cssLib.i",
-                           SWIGFLAGS=['-python', '-c++', '-Iinclude'],
-                           SWIGPATH=['..', pythonInc],
-                           CPPPATH=cppPath,
-                           SWIGCXXFILESUFFIX='_wrap.c++',
-                           SWIGOUTDIR=env.Dir("python"))
+# pybind11 Python wrapper goes to a separate library, cc file is also
+# excluded from standardModule()
+# pybind11 makes too many warnings when built with -pedantic
```
Comment (pschella) [inline core/modules/css/SConscript]: Would be useful to know what the warnings are. Perhaps something we can get fixed upstream?

File: core/modules/qmeta/qmetaPythonWrapper.cc [hunk 2]
```diff
@@ -0,0 +1,107 @@
+// Class header
```
Comment (pschella) [inline core/modules/qmeta/qmetaPythonWrapper.cc]: Missing copyright statement?

File: core/modules/qmeta/qmetaPythonWrapper.cc [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                "status"_a=std::vector<QInfo::QStatus>(),
+                "completed"_a=-1, "returned"_a=-1)
+        .def("getPendingQueries", &QMeta::getPendingQueries)
+        .def("getQueryInfo", &QMeta::getQueryInfo)
+        .def("getQueriesForDb", &QMeta::getQueriesForDb)
+        .def("getQueriesForTable", &QMeta::getQueriesForTable)
+        ;
+
+    // Exception classes
+    static py::exception<Exception> Exception(mod, "Exception");
```
Comment (pschella) [inline core/modules/qmeta/qmetaPythonWrapper.cc]: Are you sure you want to use the name of the built-in `Exception`? What if the user does an `import *`?

File: core/modules/css/SConscript (hunk 1 above)
Comment (andy-slac) [inline core/modules/css/SConscript]: Here is what appears with `-pedantic` enabled:
```
Compiling shared object build/css/cssPythonWrapper.os
In file included from /u2/salnikov/lsstsw/stack/Linux64/pybind11/master-g6f4b1f770b/include/pybind11/cast.h:13:0,
                 from /u2/salnikov/lsstsw/stack/Linux64/pybind11/master-g6f4b1f770b/include/pybind11/attr.h:13,
                 from /u2/salnikov/lsstsw/stack/Linux64/pybind11/master-g6f4b1f770b/include/pybind11/pybind11.h:36,
                 from core/modules/css/cssPythonWrapper.cc:28:
/u2/salnikov/lsstsw/stack/Linux64/pybind11/master-g6f4b1f770b/include/pybind11/pytypes.h: In lambda function:
/u2/salnikov/lsstsw/stack/Linux64/pybind11/master-g6f4b1f770b/include/pybind11/pytypes.h:1023:89: warning: ISO C++ forbids casting between pointer-to-function and pointer-to-object [-Wpedantic]
             auto destructor = reinterpret_cast<void (*)(void *)>(PyCapsule_GetContext(o));
                                                                                         ^
/u2/salnikov/lsstsw/stack/Linux64/pybind11/master-g6f4b1f770b/include/pybind11/pytypes.h: In constructor 'pybind11::capsule::capsule(const void*, void (*)(void*))':
/u2/salnikov/lsstsw/stack/Linux64/pybind11/master-g6f4b1f770b/include/pybind11/pytypes.h:1031:50: warning: ISO C++ forbids casting between pointer-to-function and pointer-to-object [-Wpedantic]
         if (PyCapsule_SetContext(m_ptr, (void *) destructor) != 0)
                                                  ^
/u2/salnikov/lsstsw/stack/Linux64/pybind11/master-g6f4b1f770b/include/pybind11/pytypes.h: In constructor 'pybind11::capsule::capsule(void (*)())':
/u2/salnikov/lsstsw/stack/Linux64/pybind11/master-g6f4b1f770b/include/pybind11/pytypes.h:1036:66: warning: ISO C++ forbids casting between pointer-to-function and pointer-to-object [-Wpedantic]
         m_ptr = PyCapsule_New(reinterpret_cast<void *>(destructor), nullptr, [](PyObject *o) {
                                                                  ^
/u2/salnikov/lsstsw/stack/Linux64/pybind11/master-g6f4b1f770b/include/pybind11/pytypes.h: In lambda function:
/u2/salnikov/lsstsw/stack/Linux64/pybind11/master-g6f4b1f770b/include/pybind11/pytypes.h:1037:92: warning: ISO C++ forbids casting between pointer-to-function and pointer-to-object [-Wpedantic]
             auto destructor = reinterpret_cast<void (*)()>(PyCapsule_GetPointer(o, nullptr));
                                                                                            ^
```
Would be nice to have upstream fixed, indeed.

File: core/modules/qmeta/qmetaPythonWrapper.cc (hunk 2 above)
Comment (andy-slac) [inline core/modules/qmeta/qmetaPythonWrapper.cc]: Will add one

File: core/modules/qmeta/qmetaPythonWrapper.cc (hunk 3 above)
Comment (andy-slac) [inline core/modules/qmeta/qmetaPythonWrapper.cc]: This is indeed a potential issue ans it all comes from C++ naming of things that were established before we started adding wrappers. We have limited number of clients of this code, all controlled by us so we can make sure that `import *` does not happen. I'll try to see if we can change Exception to something else, more Python-friendly.

Comment (andy-slac) [inline core/modules/qmeta/qmetaPythonWrapper.cc]: I renamed C++ and Python classes from Exception to QMetaError.

**Pull request merged**